code/socialDistribution/static/js/user/profile.js

```diff
@@ -1,25 +1,60 @@
-    // Enable Edit form 
-    $( "#profile_edit_button" ).click( event => {
-        event.preventDefault(); // don't submit form
-
-        // Loop all inputs
-        for (input of $("input")) {
-            // prevent username edit
-            if (input.name === 'username') {
-                continue;
-            }
-
-            // enable input
-            input.disabled = false;
+/* 
+ * Makes a GET request to the location.href and replaces the
+ * posts of the author in the author-prof-posts div by
+ * the new ones in the returned data
+ * Enable Edit form
+ * 
+ */
+
+$( "#profile_edit_button" ).click( event => {
+    event.preventDefault(); // don't submit form
+
+    // Loop all inputs
+    for (input of $("input")) {
+        // prevent username edit
+        if (input.name === 'username') {
+            continue;
         }
 
-        // set focus and cursor to first editable input end
-        firstNameInput = $('#first_name')
-        firstNameInputValue = firstNameInput.focus().val();
-        firstNameInput.val('').val(firstNameInputValue);
+        // enable input
+        input.disabled = false;
+    }
+
+    // set focus and cursor to first editable input end
+    firstNameInput = $('#first_name')
+    firstNameInputValue = firstNameInput.focus().val();
+    firstNameInput.val('').val(firstNameInputValue);
+})
+
+$('input').on('input', () => {
+    // show submit button
+    $('#profile_submit_button').removeClass('d-none');
+})
+
+
+const intervalInMs = 30000;
+let interval = setInterval(update, intervalInMs)    // fetch every 30 seconds
+
+function update() {
+    console.log("Update");
+
+    $.ajax({
+        url: location.href,
+        type: "GET",
+        success: function (data) {
+            console.log('Success');
+            let new_posts = $(data).find('.author-prof-posts').html();
+
+            // Update posts
+            $('.author-prof-posts').html(new_posts);
+
+            configEditPostModal();
```

Comment:
Is there a reason this is called twice? Once here and once on line 60.

Reply:
I called it twice because the id of the edit post's modal contains the id of the post the modal is used for. So, like if a post gets added, it's edit post modal needs to be configured as well.
I included the other one on line 60 because when the edit post modal is opened before the callback for the ajax call was executed, the modal would disappear at some point and the user would have to refresh the page. I'm not sure if this totally fixes the issue but it seemed to go away once I included that call.